ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -24,15 +30,169 @@ int TestListBuckets(const Aws::S3Crt::ClientConfiguration &config);
+        return (1);
+    }
+
+    /* List single. */
+    objects.clear();
+    expectedResult = 1;
+    if (ret = conn.ListObjects(firstBucket, prefix, objects, batchSize, listSingle) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    /* Expected number of matches with test_list_objects_1 prefix. */
+    objects.clear();
+    expectedResult = 11;
+    if (ret = conn.ListObjects(firstBucket, prefix + "1", objects) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    /* List with 5 objects per AWS request. */
+    objects.clear();
+    batchSize = 5;
+    expectedResult = totalObjects;
+    if (ret = conn.ListObjects(firstBucket, prefix, objects, batchSize) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    /* ListSingle with 8 objects per AWS request. */
+    objects.clear();
+    expectedResult = 1;
+    if (ret = conn.ListObjects(firstBucket, prefix, objects, batchSize, listSingle) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    /* List with 8 objects per AWS request. */
+    objects.clear();
+    batchSize = 8;
+    expectedResult = totalObjects;
+    if (ret = conn.ListObjects(firstBucket, prefix, objects, batchSize) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    /* ListSingle with 8 objects per AWS request. */
+    objects.clear();
+    expectedResult = 1;
+    if (ret = conn.ListObjects(firstBucket, prefix, objects, batchSize, listSingle) != 0)
+        return (ret);
+    if (objects.size() != expectedResult) {
+        CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+        return (1);
+    }
+
+    CleanupTestListObjects(config, firstBucket, totalObjects, prefix);
+    return (0);
+}
+
+int
+CleanupTestListObjects(const Aws::S3Crt::ClientConfiguration &config, const std::string bucketName,
+  const int totalObjects, const std::string prefix)
+{
+    /* Delete objects and file at end of test. */
+    S3Connection conn(config);
+    int ret = 0;
+    for (int i = 0; i < totalObjects; i++) {
+        if (ret = conn.DeleteObject(bucketName, prefix + std::to_string(i) + ".txt") != 0)
+            std::cerr << "Error in CleanupTestListBuckets: failed to remove " << prefix
+                      << std::to_string(i) << ".txt from " << bucketName << std::endl;
+    }
+    std::remove("test_list_objects.txt");
```

Comment:
This name should not be hardcoded, should it be a variable passed to the function?